hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager/src/main/java/org/apache/hadoop/yarn/server/resourcemanager/scheduler/AbstractYarnScheduler.java

```diff
@@ -658,6 +766,14 @@ private void recoverResourceRequestForContainer(RMContainer rmContainer) {
       return;
     }
 
+    // when auto correct container allocation is enabled, there can be a case when extra containers
+    // go to expired state from allocated state. When such scenario happens do not re-attempt the
+    // container request since this is expected.
+    if (autoCorrectContainerAllocation &&
+        RMContainerState.EXPIRED.equals(rmContainer.getState())) {
+      return;
+    }
```

Comment:
What will happen if a container become expired by preemption, not by the auto correction? In this case, do we need to recover the container?

Reply:
Does preemption cause state of the container to be EXPIRED ? AFIK, The EXPIRED state indicates the allocated container is not used within a specific period.

Even if that is the case, The client would then request for additional containers if its request is not satisfied instead of YARN proactively adding the same.

Reply:
Makes sense. Thanks